scripts/breaking-change.js

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License in the project root for license information.
+
+'use strict';
+var utils = require('../test/util/utils'),
+  path = require('path'),
+  fs = require('fs'),
+  os = require('os'),
+  exec = require('child_process').exec,
+  execSync = require('child_process').execSync,
+  oad = require('oad');
+
+  // This map is used to store the mapping between files resolved and stored location
+var resolvedMapForNewSpecs = {};
+let outputFolder = path.join(os.tmpdir(), "resolved");
+// Used to enable running script outside TravisCI for debugging
+let isRunningInTraviCI = process.env.MODE === 'BreakingChange' && process.env.PR_ONLY === 'true';
+
+/**
+ * Compares old and new specifications for breaking change detection.
+ *
+ * @param {string} oldSpec Path to the old swagger specification file.
+ *
+ * @param {string} newSpec Path to the new swagger specification file.
+ */
+function runOad(oldSpec, newSpec) {
+  if (oldSpec === null || oldSpec === undefined || typeof oldSpec.valueOf() !== 'string' || !oldSpec.trim().length) {
+    return Promise.reject(new Error('oldSpec is a required parameter of type "string" and it cannot be an empty string.'));
+  }
+
+  if (newSpec === null || newSpec === undefined || typeof newSpec.valueOf() !== 'string' || !newSpec.trim().length) {
+    return Promise.reject(new Error('newSpec is a required parameter of type "string" and it cannot be an empty string.'));
+  }
+
+  console.log(`Old Spec: "${oldSpec}"`);
+  console.log(`New Spec: "${newSpec}"`);
+
+  return Promise.resolve(oad.compare(oldSpec, newSpec, { consoleLogLevel: 'warn', json: true }));
+}
+
+/**
+ * Processes the given swagger and stores the resolved swagger on to disk
+ *
+ * @param {string} swaggerPath Path to the swagger specification file.
+ */
+function processViaAutoRest(swaggerPath) {
+  if (swaggerPath === null || swaggerPath === undefined || typeof swaggerPath.valueOf() !== 'string' || !swaggerPath.trim().length) {
+    return Promise.reject(new Error('swaggerPath is a required parameter of type "string" and it cannot be an empty string.'));
+  }
+
+  console.log(`Processing via AutoRest...`);
+
+  let outputFileNameWithExt = swaggerPath.split('/').pop();
+  let outputFileNameWithoutExt = outputFileNameWithExt.split('.')[0];
+  let autoRestCmd = `autorest --input-file=${swaggerPath} --output-artifact=swagger-document.json --output-file=${outputFileNameWithoutExt} --output-folder=${outputFolder}`;
+
+  console.log(`Executing : ${autoRestCmd}`);
+
+  try {
+    let result = execSync(`${autoRestCmd}`, { encoding: 'utf8', maxBuffer: 1024 * 1024 * 64 });
```

Comment:
not sure as I derived it from linter https://github.com/Azure/azure-rest-api-specs/blob/current/test/linter.js#L20

I think idea was to keep something to make sure we see when things goes over some number.